File: javascript/com/github/devginie/codility/lesson3/FrogJmp.js
/**
 * https://codility.com/programmers/task/frog_jmp/
 * Count minimal number of jumps from position X to Y.
 *
 * @author devginie
 * score: 100
 */

function solution(X, Y, D) {
    // write your code in JavaScript (Node.js 4.0.0)
    return Math.ceil((Y - X) / D);
}

File: java/org/tinywind/codility/lesson3/FrogJmp.java
package org.tinywind.codility.lesson3;

/**
 * https://codility.com/programmers/task/frog_jmp/
 * Count minimal number of jumps from position X to Y.
 *
 * @author tinywind
 * score: 100
 */
public class FrogJmp {
    public static void main(String[] args) {
        p(10, 85, 30);
    }

    private static void p(int X, int Y, int D) {
        System.out.println("X(" + X + ") Y(" + Y + ") D(" + D + ") : " + solution(X, Y, D));
    }

    private static int solution(int X, int Y, int D) {
        return (int) Math.ceil(((double)Y - (double)X) / (double)D);
    }
}
